pkg/aave.go

```diff
@@ -106,7 +107,35 @@ type AaveOperation struct {
 	client *ethclient.Client
 }
 
+func isAaveChainSupported(chainID *big.Int, fork AaveProtocolFork) error {
+
+	if !IsBnb(chainID) && !IsEth(chainID) {
+		return errors.New("only eth and bnb chains are supported")
+	}
+
+	if IsBnb(chainID) && fork == AaveProtocolForkSpark {
+		return errors.New("Spark finance is not supported on Bnb chain. Only Ethereum")
+	}
+
+	return nil
+}
+
 func NewAaveOperation(client *ethclient.Client, chainID *big.Int, fork AaveProtocolFork) (*AaveOperation, error) {
+
+	if err := isAaveChainSupported(chainID, fork); err != nil {
+		return nil, err
+	}
+
+	networkID, err := client.NetworkID(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("client.NetworkID: could not fetch network id.%v", err)
```

Comment:
you could wrap the error to capture stack trace `Wrap`. I think you used it somewhere else.

Reply:
I don't think we use stack tracing wrapping anywhere. I have updated it to use the more standard `%w`